5-design-patterns/homework/src/Package.ts

```diff
@@ -0,0 +1,23 @@
+import { Shipment } from "./Shipment";
+import { SHIPMENT_TYPE, ShipmentParams } from "./interfaces";
+import { Shipper } from "./Shipper";
+
+export class Package extends Shipment {
+    private static package: Package;
+    constructor(shipmentParams: ShipmentParams, shipper: Shipper) {
+        super(shipmentParams, shipper);
+    }
+
+    public static getInstance(shipmentParams: ShipmentParams, shipper: Shipper): Package {
+        if (!Package.package) {
+            Package.package = new Package(shipmentParams, shipper);
+        }
+
+        return Package.package;
+    }
+
+    ship(): string {
+        return `${super.ship()} Cost = ${this.shipper.getCost(this.shipmentParams.weight, SHIPMENT_TYPE.PACKAGE)}`;
```

Comment:
be honest, I don't see a reason to override ship function here when you can do all the same in your abstract class. Yes, you'll need to remove that `SHIPMENT_TYPE` enum, but you don't really need it, give the shipper a whole shipment and let shipper figures out which type of the shipment using `instanceof` instead of passing through this enum value.